float64 joint1
float64 joint2
float64 joint3
